Implemented Forgot-Password Page as per the Task I took in Trello

ajfranceschi opened the pull request:
Added FormInput Component,
Added and implemented Forgot Password Page
Added forgot-password route to Router Switch.

codinggilm committed:
added myself

ArnasDickus committed:
Merge pull request #22 from codinggilm/contributorRequest

added myself

lucierabahi committed:
Merge pull request #26 from zero-to-mastery/development

Merging development branch to master branch

ajfranceschi committed:
Merge remote-tracking branch 'upstream/development' into development

# Conflicts:
#	resolved CONTRIBUTORS.md

ajfranceschi committed:
added Link to forgot-password page at end of form and styled the button.

ajfranceschi committed:
Changed forgotPassword link classes

ajfranceschi committed:
Added ForgotPassword page to Router

ajfranceschi committed:
implemented Forgot-password page.

ajfranceschi committed:
Merge branch 'development' of github.com:zero-to-mastery/visual-music into development

* 'development' of github.com:zero-to-mastery/visual-music:
  changed the Signup component to App component in index.js file
  Done improvement according to given instruction
  Added signup component

# Conflicts:
#	CONTRIBUTORS.md

ajfranceschi committed:
Merge branch 'development' into freshKenobi

* development:
  changed the Signup component to App component in index.js file
  Done improvement according to given instruction
  Added signup component

ArnasDickus commented:
@francofle Why did you close it?

ajfranceschi commented:
I just saw that we weren't supposed to use pure css.

ajfranceschi commented:
I'll reopen and you let me know if it's ok, or you need me to do changes.

ArnasDickus commented:
@francofle You don't need to close PR, in order to add some changes. Leave this PR open. If you need to add any changes. Make changes locally, commit, them, and write `git push origin freshKenobi` again

ArnasDickus commented:
@francofle 
Travis test failed, [travis result](https://travis-ci.org/zero-to-mastery/visual-music/builds/595241111?utm_source=github_status&utm_medium=notification)

./src/components/FormInput/formInput.component.jsx
  Line 5:  Useless constructor  no-useless-constructor
If you not adding state remove constructor.

ajfranceschi committed:
Changed FormInput component from Class to Functional component.

ajfranceschi commented:
@ArnasDickus changed FormInput component from class to functional component

ArnasDickus commented on src/components/FormInput/formInput.component.jsx:
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import './formInput.styles.scss';
+
+const FormInput = ({labelText, placeholder, onEmailInputChange, ...otherInputProps}) => {
```
Why did you added ...otherInputProps?

ajfranceschi replied:
in case someone else might use the form input elsewere in the app and they need to add more classes to the input.

ArnasDickus replied:
I guess that makes sense. You can leave it.

ArnasDickus reviewed (changes requested):
Hello, thanks for adding PR. Here are my concerns:
1) PR is not formatted. To format run `npm run format`
2) All scss needs to be enclosed. So rename scss file with module. For example style.module.scss
3) I liked the submit button animation.

Other than that solid PR.

ajfranceschi committed:
Renamed style files to componentName.module.scss.
Changed import statements in respective jsx files and className assignments to match modular scss.

ajfranceschi commented:
@ArnasDickus, renamed the style files as requested and updated the import statement to match modular scss.

ajfranceschi committed:
Button.js: added ...otherProps to the component props for reusability (for example: to be able to send the onClick prop)
formInput.component.jsx:  removed the type so it could be sent when created via props.
TopNav.js:  Linked SignUp button to /signup route
forgotPasswordPage.component.jsx:  removed unnecessary console.logs && added boilerplate functionality for submit button clicked.
AppRouter.js => added /signup route

ajfranceschi committed:
Changed form SASS colors to match signup form colors.

ajfranceschi committed:
Added Reset-Password Page component and sass module.
AppRouter.js: Added /reset-password route and reformatted with Prettier.
formInput.component.jsx: reformatted with Prettier.
formInput.module.scss: reformatted with Prettier && changed input/label styles to add some space between them when multiple FormInput components are rendered.
forgotPasswordPage.component.jsx: reformatted with Prettier.
forgotPasswordPage.module.scss: reformatted with Prettier.

ajfranceschi committed:
resetPasswordPage.component.scss:  removed placeholders from input fields.
formInput.module.scss: changed input text color.

ajfranceschi committed:
Commit after running `npm run format`

ArnasDickus reviewed (approved):
Alright, looks good. Good work.
